src/Swarm/Game/Step.hs

```diff
@@ -1512,6 +1534,13 @@ execConst c vs s k = do
     when (isCardinal d) $ hasCapabilityFor COrient (TDir d)
     let nextLoc = loc ^+^ applyTurn d (orient ? zero)
     (nextLoc,) <$> entityAt nextLoc
+
+  ensureEquipped :: HasRobotStepState sig m => Text -> m Entity
+  ensureEquipped itemName = do
+    inst <- use installedDevices
+    listToMaybe (lookupByName itemName inst)
+      `isJustOrFail` ["What is", indefinite itemName <> "?"]
```

Comment:
This is nitpicky, but elsewhere I think we have reserved an error message like `What is a tree?` for situations where the robot doesn't even know about an item.  In this case the item is not equipped, but we might still have some in our inventory, or we might know about it even if we don't have any right now.  If we know about the item but don't have one currently equipped I would expect a message more like `You don't have a tree equipped.`

Reply:
fixed